web/packages/teleport/src/services/bot/types.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export type Bot = {
+  name: string;
+  roles: string[];
+  traits: string[];
+};
+
+export type FetchBotsRequest = {
+  signal?: AbortSignal;
```

Comment:
The `AbortSignal` here isn't really part of the request - I think this should be it's own separate argument to `fetchBots`, and as this request is empty otherwise, the only argument